### ISSUE TITLE
code in table styling improvement

### ISSUE THREAD

**Issue by janicecmt** [asciidoctor/asciidoctor#2173]
If you have. a table with a column containing code, the default styling is jarring because the code blends into the background in the rows with grey backgrounds but uses its own partial grey background block in the rows with white backgrounds (see attached screenshot).

It would nice to either turn off code backgrounds inside tables automatically or provide an easy option to do so (maybe a column/cell style "c" for code?).
<img width="392" alt="codebackgroundintables" src="https://cloud.githubusercontent.com/assets/24766266/25620456/6f00561a-2f1c-11e7-87c2-2fcb6f9f82ef.png">

**Comment by mojavelinux** [comment]
This is definitely something that needs to be controlled by the stylesheet as it's purely a presentation issue. We wouldn't want to introduce a special cell style as that would be mixing up content and presentation.

Keep in mind that the default stylesheet is just that, a default, intended for prototyping. There are hundreds of permutations of blocks that the stylesheet doesn't account for (otherwise it would be huge and hard to test). This is where you are encouraged to customize the stylesheet for your own needs.

There are two ways you can do that. First, you might decide to turn off the table striping, a style hint that is planned in 1.5.6. You could also just disable the striping in your own CSS.

```css
table tr:nth-of-type(even) {
  background: none;
}
```

The other approach would be to style listing blocks that are in those striped rows using something like:

```css
table tr:nth-of-type(even) .listingblock pre {
  background: #fff;
}
```

Although this seems simple enough to change, it's actually really complicated because the background color is controlled by the syntax highlighter and we support many different syntax highlighters with many different external stylesheets. So it really has to be something you control or else we risk breaking the color scheme applied by the syntax highlighter.

**Comment by mojavelinux** [comment]
Of course, an even simpler workaround is to use a different syntax highlighting theme. Then, the background color will be different.

**Comment by mojavelinux** [comment]
It will be included in 1.5.7, which I'm actively working on getting wrapped up. A rough estimate is by the end of the month, but it could be sooner.
